src/components/Modal/ConfirmationModal.tsx

```diff
@@ -0,0 +1,129 @@
+'use client'
+
+import { ReactNode, useRef } from 'react'
+import { motion, AnimatePresence } from 'motion/react'
+import { createPortal } from 'react-dom'
+import modalBg from './images/modal-bg.svg'
+import { cn } from '@/lib/utils'
+import { HeaderTitle } from '../Typography'
+import { Button } from '../Button'
+import { useClickOutside } from '@/shared/hooks/useClickOutside'
+
+// transition animation duration
+const duration = 0.3
+
+interface ConfirmationModalProps {
+  /**
+   * Modal title displayed at the top
+   */
+  title?: string
+  /**
+   * Pass the main modal text to the `children` prop
+   */
+  children?: ReactNode
+  /**
+   * Called when `I Agree` button is clicked
+   */
+  onAccept?: () => void
+  /**
+   * Called when `I Disagree` button is clicked
+   */
+  onDecline?: () => void
+  /**
+   * When defined, replaces accept/decline buttons
+   */
+  buttons?: ReactNode
+  /**
+   * Controls whether the modal is visible
+   */
+  isOpen: boolean
+  /**
+   * Callback triggered when the modal is closed
+   */
+  onClose?: () => void
+  /**
+   * Test identifier for unit and integration tests
+   */
+  'data-testid'?: string
+}
+
+/**
+ * Generic reusable animated modal component.
+ * Can be used for disclaimers, warnings, confirmations, or other modal dialogs.
+ *
+ * It is recommended to use this component together with the `useModal` hook
+ * for better state management and handling open/close logic.
+ */
+export function ConfirmationModal({
+  title,
+  children,
+  onClose,
+  'data-testid': dataTestId,
+  buttons,
+  onAccept,
+  onDecline,
+  isOpen = false,
+}: ConfirmationModalProps) {
+  const modalRef = useRef<HTMLDivElement>(null)
+  useClickOutside([modalRef], () => onClose && onClose())
```

Comment:
You can rewrite as `onClose?.()` and get rid of that conditional operation. But I would say that this function should be always defined, it should close modal by default.

Reply:
agree

Reply:
regarding onAccept and onDecline - they are NOT going to close the modal. Modal triggering should be performed outside the modal, specifically from the hook `useModal`. So here, I am not gonna change anything, sorry

Reply:
I agree that `onAccept` must be defined outside. But generally the negative action of any modal is closing it. It's reasonable to define `onDecline` to close the modal by default. But if you have a strong opinion against it, it's fine though.

Reply:
I made onClose prop required

Reply:
My point is, I dont know in advance what opening/closing logic will the user of the modal need. Maybe he will need to keep the modal on screen. And you are advising to decide it for him in advance. You see